src/main/java/au/com/acegi/xmlformat/XmlCheckPlugin.java

```diff
@@ -0,0 +1,62 @@
+/*-
+ * #%L
+ * XML Format Maven Plugin
+ * %%
+ * Copyright (C) 2011 - 2019 Acegi Technology Pty Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package au.com.acegi.xmlformat;
+
+import static au.com.acegi.xmlformat.FormatUtil.needsFormatting;
+import java.io.File;
+import java.io.IOException;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.logging.Log;
+import static org.apache.maven.plugins.annotations.LifecyclePhase.PROCESS_SOURCES;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.dom4j.DocumentException;
+import org.dom4j.io.OutputFormat;
+
+/**
+ * Finds the XML files in a project and only check them: no files are changed,
+ * but the build will fail if any file does not follow the formatting
+ * conventions.
+ */
+@Mojo(name = "xml-check", defaultPhase = PROCESS_SOURCES)
```

Comment:
The default phase is debatable, but I feel like if it's going to fail my build, I'd rather have it happen sooner.